File: flask-api/src/serializers/service_rate.py
from marshmallow import Schema, fields

class ServiceRateSchema(Schema):
    service_id = fields.Integer(required=True)
    unit = fields.Float(required=True) 
    amount = fields.Float(required=True) 
    duration = fields.Float(required=True) 
    supply_markup = fields.Float(required=True) 
    overhead_markup = fields.Float(required=True) 
    misc_markup = fields.Float(required=True) 
    
    class Meta:
        ordered = True

File: php-soapservice/src/Entity/ServiceRate.php
<?php

namespace Application\Entity;

class ServiceRate extends ActiveRecord
{
    const TABLE_NAME = 'service_rates';

    public int $service__id;

    public float $unit;

    public float $amount;

    public float $duration;

    public float $supply_markup;

    public float $overhead_markup;

    public float $misc_markup;
    

    public function jsonSerialize()
    {
        return [
            'id' => $this->id,
            'service_id' => $this->service_id,
            'unit' => $this->unit,
            'amount' => $this->amount,
            'duration' => $this->duration,
            'supply_markup' => $this->supply_markup,
            'overhead_markup' => $this->overhead_markup,
            'misc_markup' => $this->misc_markup
        ];
    }
}
